### DISCUSSION TITLE
Welcome to OpenTCS-4 Discussions!

### DISCUSSION THREAD

**Discussion by touchmii** [touchmii/OpenTCS-4#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Javac0** [comment]
老哥，你这怎么运行起来，能否有详细的文档说明，我这一直运行不起来，持久化到内核中一直失败。jettyRunWar运行没有出现网页呀

**Comment by antsgod** [comment]
老哥，能不能分享下opentcs怎么学习的，学了很久，感觉一直没入门

**Comment by touchmii** [comment]
没有接触java spring这类注入框架可能不好理解opentcs中juice的作用，搞清楚java 反射、代理这些代理再去深入了解tcs的源码。
我开始不接了注入，改个适配器走了不少弯路。要理解tcs内部运行机制光看代码可能不够直观，可以下断点，一步一步看程序
如何运行的，我好像有写一些笔记，有空我更新到博客上面去。jettyrunwar那个webclient不用研究了，没有实际用途，只是测试
能否提供rap这个框架开发web界面，目前打算使用react写。

**Comment by antsgod** [comment]
十分感谢。其实我是做后端的，日常开发工作就是用java,spring。只是接触opentcs后，不知道从哪里下手。。。。。网上的资料太少了，你是不是就看官网的手册就看懂了呢



***@***.***
 
From: Benoni Jiang
Date: 2022-10-20 15:39
To: touchmii/OpenTCS-4
CC: antsgod; Comment
Subject: Re: [touchmii/OpenTCS-4] Welcome to OpenTCS-4 Discussions! (Discussion #8)
没有接触java spring这类注入框架可能不好理解opentcs中juice的作用，搞清楚java 反射、代理这些代理再去深入了解tcs的源码。 
我开始不接了注入，改个适配器走了不少弯路。要理解tcs内部运行机制光看代码可能不够直观，可以下断点，一步一步看程序 
如何运行的，我好像有写一些笔记，有空我更新到博客上面去。jettyrunwar那个webclient不用研究了，没有实际用途，只是测试 
能否提供rap这个框架开发web界面，目前打算使用react写。

**Comment by touchmii** [comment]
那应该没有什么难度呀，我接触tcs前都没有用过java，只会c和python，属于散装程序员。通过tcs学习了到了很多设计模式，他的源码写的比较规范，我把笔记更新到csdn了，https://blog.csdn.net/touchmii?type=lately，希望对你有帮助，有个两个qq群你可以加一下，搜索opentcs就能找到。

**Comment by antsgod** [comment]
感谢，感谢，感谢。qq号好像搜不到。。。



***@***.***
 
From: Benoni Jiang
Date: 2022-10-20 17:29
To: touchmii/OpenTCS-4
CC: antsgod; Comment
Subject: Re: [touchmii/OpenTCS-4] Welcome to OpenTCS-4 Discussions! (Discussion #8)
那应该没有什么难度呀，我接触tcs前都没有用过java，只会c和python，属于散装程序员。通过tcs学习了到了很多设计模式，他的源码写的比较规范，我把笔记更新到csdn了， 
https://blog.csdn.net/touchmii?type=lately 
，希望对你有帮助，有个两个qq你可以加一下，搜索opentcs就能找到。
